code/datums/action.dm

```diff
@@ -669,16 +674,31 @@
 /datum/action/cooldown
 	check_flags = NONE
 	transparent_when_unavailable = FALSE
-	// The default cooldown applied when StartCooldown() is called
+	// The default cooldown applied after the ability ends
 	var/cooldown_time = 0
 	// The actual next time this ability can be used
 	var/next_use_time = 0
+	// The default melee cooldown applied after the ability ends
+	var/melee_cooldown_time
+	// The actual next time the owner of this action can melee
+	var/next_melee_use_time = 0
 	// Whether or not you want the cooldown for the ability to display in text form
 	var/text_cooldown = TRUE
 	// Setting for intercepting clicks before activating the ability
 	var/click_to_activate = FALSE
-	// Shares cooldowns with other cooldown abilities of the same value, not active if null
+	// Shares cooldowns with other abiliies, bitflag
 	var/shared_cooldown
+	// List of prerequisite actions that are used in this sequenced ability, you cannot put other sequenced abilities in this
+	var/list/sequence_actions
+	// List of prerequisite actions that have been initialized
+	var/list/initialized_actions
+
+/datum/action/cooldown/New(Target, original = TRUE)
+	..()
+	if(isnull(melee_cooldown_time))
+		melee_cooldown_time = cooldown_time
```

Comment:
This won't always set a melee cooldown, only if melee cooldown time was not set. If you set melee cooldown time to 0, it is not considered null and so the cooldown will be nothing. The only cases where they are equal is if you set them to be equal or if the melee cooldown isnt set at all.

I could move melee cooldown times to the mob cooldown subtype but this project was set out as a way to avoid things like wizard subtypes only working for carbon mobs and being weirdly unusable on other types of mobs since they are pretty much hardcoded for only that one specific thing even for basic things like projectile firing.

I don't really like the idea of having a bunch of subtypes for stuff but I do understand so if you want me to move it to the mob cooldown type I could, but that was more meant for variable setting like shared cooldowns and a base cooldown timer which wouldn't really fit on the base type.